libs/blocks/section-metadata/section-metadata.js

```diff
@@ -71,6 +77,38 @@ export const getMetadata = (el) => [...el.childNodes].reduce((rdx, row) => {
   return rdx;
 }, {});
 
+async function handleShowMoreButton(section) {
+  const showMoreBtn = section.querySelector('.show-more-button button');
```

Comment:
You could add the button as an argument. My rule of thumb is to avoid querying for elements we create, as much as possible, since we should already have references to them.